Added Jenkinsfile to download zipkin-server

awharris opened the pull request:
- Downloads Zipkin server from artifactory
- Zipkin version set at 2.9.4 to support both v1 and v2 APIs

Commit:
Added Jenkinsfile to download zipkin-server

- Downloads Zipkin server from artifactory
- Zipkin version set at 2.9.4 to support both v1 and v2 APIs

Commit:
Added missing "stages" tag in Jenkinsfile

- Corrected pipeline syntax

willis7 commented on Jenkinsfile:
```diff
@@ -11,12 +11,14 @@ pipeline {
+        stage('Download') {
+            steps {
+                script {
+                    def server = Artifactory.server art-p-01
```
This will work for the scope of the Download stage only. If you need to perform multiple downloads then you will want to make server global. This is a good candidate to move into the shared library.

willis7 commented on Jenkinsfile:
```diff
@@ -0,0 +1,46 @@
+#!groovy
+@Library('jenkins-pipeline-shared') _
```
This isn't declarative ;-)

Commit:
Added checkout and deploy stages

- Added missing checkout stage
_ Added deploy to DEV CF stage

Commit:
Moved zipkin-server download to checkout stage

- Downloads zipkin-server jar as part of checkout stage

Commit:
Adde CF manifest

- Added basic CF manifest
- Changed download repo
- Fixed CF deployment params

Commit:
Changed artifactory repo

- Download path for zipkin=server is now jcenter-cache/io/zipkin/java/zipkin-server/2.9.4/

Commit:
Change artifactory repo

- File download set to
libs-release/io/zipkin/java/zipkin-server/2.9.4/zipkin-server-2.9.4-exec
.jar

Commit:
Fixed Artifactory download

- Added missing buildInfo
- Improved download spec to include build number

Commit:
Fixed flattening of directory structure on download

- Artifactory download spec flat is a string not boolean

Commit:
Fixed Jenkinsfile

- Previous version would not execute

Commit:
Fixed wildcards in path

- Wildcards now (*) not *

Commit:
Hardcode Zipkin version

- Hardcode Zipkin version to get it working

Commit:
Fixed manifest.yml

- Typo in file

Commit:
Added healthcheck after deployment

- Added call to zipkin-server /health endpoint to test deployment

Commit:
Refactored Jenkinsfile

- Added health check function
- Move stage reporting to functions

Commit:
Fixes when health check fails

- Fixed handling 404 to health check
- Refactoring

Commit:
Added parallel BI and SBR deployment

- Deploys to both BI and SBR CF spaces

Commit:
Fixed typo in Jenkinsfile

- Missing quotes

Commit:
Removed milestone from Jenkinsfile

- milestone not supported in parallel step

Commit:
Minor Jenkinsfile change

- Tidy up

Commit:
Set CF instances to 1

- Set instances to 1 in default CF manifest file

Commit:
Refactored CF deploy

- Now passes org and space as params to deploy function
- Similar changes for health check test

Commit:
Fixed post deploy

- Incorrect use of post

Commit:
Fixed post after health check

- Incorrect use of post

Commit:
Pipeline now deploys to DEV, TEST and PROD

- Added sequential deployment through DEV, TEST and PROD
- Refactored common deploy and health check methods

Commit:
Only deploys on master branch

- Add conditions to only deploy on master branch

Commit:
Simplified master branch checks

- Used built-in branch check condition

willis7 commented on Jenkinsfile:
```diff
@@ -0,0 +1,214 @@
+                    } else {
+                        stageFailure()
+                        error("Cannot find file ${MODULE_NAME}.jar")
+                    }
```
This is some good error checking. Would it be more useful as part of the shared library inside the deploy method?

willis7 commented on Jenkinsfile:
```diff
@@ -0,0 +1,214 @@
+                DEV_ROUTE = "${DEV}-${TEAM}-${MODULE_NAME}"
+            }
+            steps {
+                milestone(1)
```
Is it worth naming milestones to be a little clearer? `milestone(label: "begin deploy dev", ordinal: 1)`

willis7 commented on Jenkinsfile:
```diff
@@ -0,0 +1,214 @@
+            agent any
+	        when { branch 'master' }
+            steps {
+                healthCheck("${DEV}-${TEAM}-${MODULE_NAME}")
```
I was planning on doing this exact thing in the deploy step also. Nice addition

willis7 commented on Jenkinsfile:
```diff
@@ -0,0 +1,214 @@
+
+def stageSuccess() {
+    colourText("info", "Stage: ${STAGE_NAME} successful!")
+}
```
shared pipeline?

willis7 reviewed (dismissed):
There's some good stuff that could make its way to the shared library. I also think we need to agree and pin stage names and use them everywhere.

Commit:
Set agents in Jenkinsfile

Commit:
Added missing SVC_NAME in Jenkinsfile
